src/governance/governance.cpp

```diff
@@ -675,7 +675,7 @@ std::optional<const CSuperblock> CGovernanceManager::CreateSuperblockCandidate(i
 std::optional<const CGovernanceObject> CGovernanceManager::CreateGovernanceTrigger(const std::optional<const CSuperblock>& sb_opt, CConnman& connman,
                                                                                    const CActiveMasternodeManager* const mn_activeman)
 {
-    if (!fMasternodeMode) return std::nullopt;
+    if (mn_activeman == nullptr) return std::nullopt;
```

Comment:
let's move this check out CreateGovernanceTrigger.

1. change signature:
```
std::optional<const CGovernanceObject> CGovernanceManager::CreateGovernanceTrigger(const std::optional<const CSuperblock>& sb_opt, CConnman& connman,
                                                                                   const CActiveMasternodeManager& mn_activeman)
```

2: change call:
```diff
-const auto sb_opt = CreateSuperblockCandidate(pindex->nHeight);
-const auto trigger_opt = CreateGovernanceTrigger(sb_opt, connman, mn_activeman);
-VoteGovernanceTriggers(trigger_opt, connman, mn_activeman);
+if (mn_activeman) {
+    const auto sb_opt = CreateSuperblockCandidate(pindex->nHeight);
+    const auto trigger_opt = CreateGovernanceTrigger(sb_opt, connman, *mn_activeman);
+    VoteGovernanceTriggers(trigger_opt, connman, *mn_activeman);
+}
```

Reply:
Checks out, seems to be used only there. Opportunistic dereferencing helps us get rid of some asserts, great catch!